src/main/java/com/hillel/items_exchange/controller/UserController.java

```diff
@@ -53,20 +55,18 @@
     }
 
     @GetMapping("/child")
-    public ResponseEntity<List<ChildDto>> getChildren(Principal principal) {
+    public ResponseEntity<List<ChildUpdateDto>> getChildren(Principal principal) {
         return new ResponseEntity<>(userService.getChildren(getUser(principal.getName())), HttpStatus.OK);
     }
 
     @PostMapping("/child")
     public ResponseEntity<HttpStatus> addChildren(@RequestBody
-                                                  @Size(min = 1, message = "{exception.invalid.dto}")
-                                                          List<@Valid ChildDto> childrenDto,
+                                                  @Size(min = 1, max = 10, message = "{exception.invalid.dto}")
+                                                          List<@Valid ChildAddDto> childrenDto,
                                                   Principal principal) {
-        if (childrenDto.stream().anyMatch(dto -> dto.getId() > 0)) {
-            throw new IllegalIdentifierException(
-                    getExceptionMessageSourceWithAdditionalInfo(
-                            "exception.illegal.id",
-                            "Zero ID is expected"));
+        if(childrenDto.size() + userService.getChildren(getUser(principal.getName())).size() >= 10) {
```

Comment:
Maybe something like this for better understanding? But it's up to you)))
```suggestion
             int amountOfChildren = childrenDto.size() + userService.getChildren(getUser(principal.getName())).size();
        if (amountOfChildren > 10) {
```

Reply:
@drw85 have you tried to add Size annotation to the DTO and validate children size with it?

Reply:
@rpkyrych It will resolve the issue of the incoming size, but we need total amount of user children to be lees than 10. Right?

Reply:
So you could use Max(10) to allow values less than allowed.

Reply:
I checked it with clean eyes and agree with @drw85 and agreed that it would nice to have additional variable here.

Reply:
You are all right, I will add additional variable here